vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java

```diff
@@ -0,0 +1,123 @@
+package ca.bc.gov.nrs.vdyp.common;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Type union of a value and an out of band marker.
+ *
+ * @author Kevin Smith, Vivid Solutions
+ *
+ * @param <Value>
+ * @param <Marker>
+ */
+public class ValueOrMarker<Value, Marker> {
+	private final boolean isMarker;
+	private final Object obj;
+
+	private ValueOrMarker(Object obj, boolean isMarker) {
+		Objects.requireNonNull(obj, "The Value or Marker of a ValueOrMarker must not be null");
+
+		this.isMarker = isMarker;
+		this.obj = obj;
+	}
+
+	/**
+	 * Is there a value
```

Comment:
@return true if and only if this ValueOrMarker is a marker.

Reply:
Yes, I got those backward.